File: 1-easy/problems/2670-find-the-distinct-difference-array/solution.py
from typing import List, Optional, Dict, Set, Tuple

class Solution:
    def distinctDifferenceArray(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        result = [0]*len(nums)
        lookup = set()
        for i in range(len(nums)):
            lookup.add(nums[i])
            result[i] = len(lookup)
        lookup.clear()
        for i in reversed(range(len(nums))):
            result[i] -= len(lookup)
            lookup.add(nums[i])
        return result

File: 1-easy/problems/2670-find-the-distinct-difference-array/solution.ts
function distinctDifferenceArray(nums: any): boolean | number | string | any {
  """
  :type nums: List[int]
  :rtype: List[int]
  """
  result = [0]*(nums).length
  lookup = set()
  for (i in Array.from({length: (nums}, (_, i) => i).length)) {
      lookup.add(nums[i])
      result[i] = (lookup).length
  lookup.clear()
  for (i in reversed(Array.from({length: (nums}, (_, i) => i).length))) {
      result[i] -= (lookup).length
      lookup.add(nums[i])
  return result;
}

export { distinctDifferenceArray };
